tests/conftest.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=redefined-outer-name,unused-argument
+"""Configuration and fixtures for unit test suite."""
+from __future__ import annotations
+
+import typing as t
+
+import pytest
+
+pytest_plugins = ['aiida.manage.tests.pytest_fixtures']  # pylint: disable=invalid-name
+
+
+def recursive_merge(left: dict[t.Any, t.Any], right: dict[t.Any, t.Any]) -> None:
+    """Recursively merge the ``right`` dictionary into the ``left`` dictionary.
+
+    :param left: Base dictionary.
+    :param right: Dictionary to recurisvely merge on top of ``left`` dictionary.
+    """
+    for key, value in right.items():
+        if (key in left and isinstance(left[key], dict) and isinstance(value, dict)):
+            recursive_merge(left[key], value)
+        else:
+            left[key] = value
+
+
+@pytest.fixture(scope='function')
+def fixture_sandbox():
+    """Return a `SandboxFolder`."""
+    from aiida.common.folders import SandboxFolder
+    with SandboxFolder() as folder:
+        yield folder
+
+
+@pytest.fixture
+def generate_calc_job(fixture_sandbox):
+    """Fixture to construct a new `CalcJob` instance and call `prepare_for_submission` for testing `CalcJob` classes.
+
+    The fixture will return the `CalcInfo` returned by `prepare_for_submission` and the temporary folder that was passed
+    to it, into which the raw input files will have been written.
+    """
+
+    def _generate_calc_job(entry_point_name, inputs=None):
+        """Fixture to generate a mock `CalcInfo` for testing calculation jobs."""
+        from aiida.engine.utils import instantiate_process
+        from aiida.manage.manager import get_manager
+        from aiida.plugins import CalculationFactory
+
+        manager = get_manager()
+        runner = manager.get_runner()
+
+        process_class = CalculationFactory(entry_point_name)
+        process = instantiate_process(runner, process_class, **inputs)
+
+        calc_info = process.prepare_for_submission(fixture_sandbox)
+
+        return calc_info, fixture_sandbox
+
+    return _generate_calc_job
+
+
+@pytest.fixture
+def generate_structure():
+    """Return a ``StructureData`` representing a water molecule."""
+    from ase.build import molecule
+    from aiida.orm import StructureData
```

Comment:
I am always unsure about this. Is the best practice to import directly like this or via

```python
StructureData = DataFactory('core.structure')
```
? 
Are there any performance implications?

Reply:
I have to say that I don't apply it myself very rigorously. Long time ago we recommended to always use `DataFactory`, even for types in `aiida-core`, because then it would not break if we change the module location. But then again, we make sure to never break this (at least not outside of major version bumps). So I tend to just import from `aiida.orm` now

Reply:
Okay, thank you. Well, the the `core.` prefix refactor kind of broke the whole promise of DataFactory approach tbh.

Reply:
Are you sure? I think `DataFactory('structure')` still works in v2.0. It emits a deprecation warning, but we made specifically sure that it didn't break. The old entry points are automatically forwarded to the new `core.` prefixed ones. This way, you can keep `DataFactory('structure')` and it works in both v1.x and v2.x

Reply:
Yes, sorry for being unclear, it's exactly how you describe. My point was that presumably backwards compatibility will be removed in the future v3 version, while the whole point of entry points (imo) is that they should be stable. :-) All good, I understand the reasons for this decision.